youtube_dl/extractor/imgur.py

```diff
@@ -13,7 +13,7 @@
 
 
 class ImgurIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:i\.)?imgur\.com/(?!gallery)(?P<id>[a-zA-Z0-9]+)'
+    _VALID_URL = r'https?://(?:i\.)?imgur\.com/(gallery/)?(?P<id>[a-zA-Z0-9]{6,})'
```

Comment:
http://imgur.com/gallery/oWeAMW2 should also be handled, hence the change `(?!gallery)` -> `(gallery/)?`.
Also for single videos the id is of length greater than 5 (generally 7), so incorporated that.